yt_dlp/extractor/lastfm.py

```diff
@@ -0,0 +1,158 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    int_or_none
+)
+
+
+class LastFMPlaylistBaseIE(InfoExtractor):
+    def _extract_current_page(self, url):
+        return int_or_none(self._search_regex(
+            r'\bpage=(\d+)', url, 'page', default=None))
+
+    def _extract_last_page(self, webpage):
+        last_page = None
+        pagination_list = self._search_regex(
+            r'(?s)class="[^"]*pagination-list[^"]*"[^>]*>(.+?)</ul>', webpage, 'pagination_list', fatal=False,
+            default=None)
+
+        if pagination_list:
+            page_numbers = re.findall(r'(\d+)', pagination_list)
+            last_page = max([int_or_none(page_number) for page_number in page_numbers])
+
+        return last_page
+
+    def _extract_entries(self, webpage):
+        tbody = self._search_regex(
+            r'(?s)tbody\s+data-playlisting-add-entries[^>]*>(.+?)</tbody>', webpage, 'tbody')
+        return [
+            self.url_result(player_url, 'Youtube')
+            for player_url in re.findall(r'href="([^"]+youtube.com/watch[^"]+)"', tbody)
+        ]
+
+    def _download_page(self, url, current_page_number, last_page_number, playlist_id):
+        note = 'Downloading page'
+        query = {}
+
+        if last_page_number:
+            note = '%s %d of %d' % (note, current_page_number, last_page_number)
+            query['page'] = current_page_number
+
+        return self._download_webpage(url, playlist_id, note, query=query)
+
+    def _entries(self, url, playlist_id):
+        webpage = self._download_webpage(url, playlist_id)
+        current_page_number = self._extract_current_page(url)
+        last_page_number = self._extract_last_page(webpage)
+        start_page_number = current_page_number or 1
+        end_page_number = last_page_number if last_page_number is not None else start_page_number
```

Comment:
y is this var even needed?

Reply:
can be omitted but its such a small change that does not add to or take away much from the program in terms of performance. also i think this var increases readability